gapir/cc/interpreter.cpp

```diff
@@ -302,46 +591,104 @@ Interpreter::Result Interpreter::load(uint32_t opcode) {
     GAPID_WARNING("Error: load not readable address %p", address);
     return ERROR;
   }
-  mStack.pushFrom(type, address);
-  return mStack.isValid() ? SUCCESS : ERROR;
+
+  switch (type) {
+    case BaseType::Bool:
+      mStack.push(*(bool*)address);
+      break;
+
+    case BaseType::Int8:
+      mStack.push(*(int8_t*)address);
+      break;
+
+    case BaseType::Int16:
+      mStack.push(*(int16_t*)address);
+      break;
+
+    case BaseType::Int32:
+      mStack.push(*(int32_t*)address);
+      break;
+
+    case BaseType::Int64:
+      mStack.push(*(int64_t*)address);
+      break;
+
+    case BaseType::Uint8:
+      mStack.push(*(uint8_t*)address);
+      break;
+
+    case BaseType::Uint16:
+      mStack.push(*(uint16_t*)address);
+      break;
+
+    case BaseType::Uint32:
+      mStack.push(*(uint32_t*)address);
+      break;
+
+    case BaseType::Uint64:
+      mStack.push(*(uint64_t*)address);
+      break;
+
+    case BaseType::Float:
+      mStack.push(*(float*)address);
+      break;
+
+    case BaseType::Double:
+      mStack.push(*(double*)address);
+      break;
+
+    case BaseType::AbsolutePointer:
+      mStack.push(*(void**)address);
+      break;
+
+    case BaseType::ConstantPointer:
+      mStack.push(Stack::ConstantPointer(*(uint32_t*)address));
+      break;
+
+    case BaseType::VolatilePointer:
+      mStack.push(Stack::VolatilePointer(*(uint32_t*)address));
+      break;
+
+    default:
+      GAPID_FATAL("Unhandled type in Load");
+      break;
+  }
+
+  return SUCCESS;
 }
 
 Interpreter::Result Interpreter::pop(uint32_t opcode) {
   mStack.discard(extract26bitData(opcode));
-  return mStack.isValid() ? SUCCESS : ERROR;
+  return SUCCESS;
```

Comment:
If we are just going to ALWAYS return SUCCESS, why even bother with a return?
Same below